pkg/driverbuilder/builder/flatcar.go

```diff
@@ -23,66 +22,68 @@ func init() {
 type flatcarTemplateData struct {
 	commonTemplateData
 	KernelDownloadURL string
-	FlatcarVersion    string
-	FlatcarChannel    string
-	KernelConfigURL   string
 }
 
 // flatcar is a driverkit target.
 type flatcar struct {
-	version   string
-	info      *flatcarReleaseInfo
-	kconfUrls []string
+	info *flatcarReleaseInfo
 }
 
-func (f flatcar) Name() string {
+func (f *flatcar) Name() string {
 	return TargetTypeFlatcar.String()
 }
 
-func (f flatcar) TemplateScript() string {
+func (f *flatcar) TemplateScript() string {
 	return flatcarTemplate
 }
 
-func (f flatcar) URLs(_ Config, kr kernelrelease.KernelRelease) ([]string, error) {
-	if kr.Extraversion != "" {
-		return nil, fmt.Errorf("unexpected extraversion: %s", kr.Extraversion)
-	}
-
-	// convert string to int
-	if kr.Version < 1500 {
-		return nil, fmt.Errorf("not a valid flatcar release version: %d", kr.Version)
-	}
-	f.version = kr.Fullversion
-
-	var err error
-	f.info, err = fetchFlatcarMetadata(kr)
-	if err != nil {
-		return nil, err
-	}
-
-	f.kconfUrls, err = getResolvingURLs(fetchFlatcarKernelConfigURL(kr.Architecture, f.info.Channel, kr.Fullversion))
-	if err != nil {
+func (f *flatcar) URLs(_ Config, kr kernelrelease.KernelRelease) ([]string, error) {
+	if err := f.fillFlatcarInfos(kr); err != nil {
 		return nil, err
 	}
 	return fetchFlatcarKernelURLS(f.info.KernelVersion), nil
 }
 
-func (f flatcar) TemplateData(c Config, _ kernelrelease.KernelRelease, urls []string) interface{} {
+func (f *flatcar) TemplateData(c Config, kr kernelrelease.KernelRelease, urls []string) interface{} {
+	// This happens when `kernelurls` option is passed,
+	// therefore URLs() method is not called.
+	if f.info == nil {
```

Comment:
Avoid dereferencing a nil pointer. We need the target GCC to be loaded in any case.

Moreover, by moving the `setGCCVersion` call inside `toTemplateData` we enforce the GCC version to be set last, after the builder has done everything it needs to eventually customize GCCVersion (in this case, f.info loading).